### PR TITLE
1221 cherry pullrequest

### PR DESCRIPTION
**Pull request by inDigiNeous**
Implement support for THREE.MeshLine
Adds new plugins 

* three_meshline_material
* three_meshline_mesh

### CODE REVIEW

File: browser/plugins/three_meshline_material.plugin.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+ThreeMeshLineMaterialPlugin.prototype.state_changed = function(ui) {
+	// if ui is null, first time called, bind the resize
+	if (!ui) {
+		E2.core.on('resize', this.resize.bind(this))
+	}
+}
+
+ThreeMeshLineMaterialPlugin.prototype.update_input = function(slot, data) {
+	// Need to update the params also, used to recreate the material
+	this.params[slot.name] = data
```
Comment (kkananen) [inline browser/plugins/three_meshline_material.plugin.js]: you could call Plugin.prototype.update_input()
that fills in this.inputValues[] 
then you wouldn't need this.params at all really

File: browser/plugins/three_meshline_mesh.plugin.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
+(function() {
+	var createThreeMeshRoot = function() {
+		return new THREE.Object3D()
+	}
+
+	var createThreeMeshLineMesh = function(geom, mats) {
+		// Have to convert the bufferGeometry to THREE.Geometry(), as the MeshLine only
+		// accepts THREE.Geometry()
+		var set_geom;
```
Comment (kkananen) [inline browser/plugins/three_meshline_mesh.plugin.js]: would be good to be consistent with naming conventions regards camelCasing, i.e. setGeom

File: browser/vendor/THREE.MeshLine.js [hunk 3]
```diff
@@ -0,0 +1,430 @@
+THREE.MeshLine = function() {
```
Comment (kkananen) [inline browser/vendor/THREE.MeshLine.js]: This is an extra copy of THREE.MeshLine.js. Also the real copy should really be under /vendor/spite not vendor/three

File: browser/vendor/three/THREE.MeshLine.js [hunk 4]
```diff
@@ -0,0 +1,430 @@
+THREE.MeshLine = function() {
```
Comment (kkananen) [inline browser/vendor/three/THREE.MeshLine.js]: This file should be under vendor/spite

File: browser/vendor/three/THREE.MeshLine.js [hunk 5]
```diff
@@ -0,0 +1,430 @@
...
+		if ( raycaster.ray.intersectsSphere( sphere ) === false ) return;
+
+		inverseMatrix.getInverse( matrixWorld );
+		ray.copy( raycaster.ray ).applyMatrix4( inverseMatrix );
+
+		var vStart = new THREE.Vector3();
+		var vEnd = new THREE.Vector3();
+		var interSegment = new THREE.Vector3();
+		var interRay = new THREE.Vector3();
+		var step = this instanceof THREE.LineSegments ? 2 : 1;
```
Comment (kkananen) [inline browser/vendor/three/THREE.MeshLine.js]: when would this ever be THREE.LineSegments?

**Comment by kschzt** [comment]
lgtm

**Pull request merged**